### ISSUE TITLE
Suggest removal of stray `;` after `struct` definition

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#46186]
For the given code

```rust
struct Struct {
    a: usize,
};
```

the output is only

```
error: expected item, found `;`
  --> file.rs:15:2
   |
15 | };
   |  ^
```

where it should suggest removing the semicolon (as a suggestion so the RLS does the right thing)

```
error: expected item, found `;`
  --> file.rs:15:2
   |
15 | };
   |  ^ consider removing this semicolon
```

This is an edge situation, but I feel it is a common enough typo I've made and that is very likely to happen to newcomers.

**Comment by raventid** [comment]
I'll take this one :)

**Comment by est31** [comment]
Newcomers from C++ are likely to hit this error.

**Comment by colinmarsh19** [comment]
@jjolly @raventid I think I almost have a working solution to this. Will make a pull request when I compile in test.

**Comment by colinmarsh19** [comment]
This can be closed due to #46312 and #46258.

**Comment by kennytm** [comment]
Closing as fixed in #46258. Thanks for contribution!
